presto-main/src/main/java/com/facebook/presto/server/HttpRequestSessionContext.java

```diff
@@ -103,6 +105,7 @@ public HttpRequestSessionContext(HttpServletRequest servletRequest)
         identity = new Identity(user, Optional.ofNullable(servletRequest.getUserPrincipal()));
 
         source = servletRequest.getHeader(PRESTO_SOURCE);
+        traceToken = Optional.ofNullable(servletRequest.getHeader(PRESTO_TRACE_TOKEN));
```

Comment:
Wrap the `getHeader()` call in `trimEmptyToNull()` so we convert empty string to empty optional